saleor/checkout/models.py

```diff
@@ -250,6 +250,30 @@ def get_country(self):
             self.set_country(country_code, commit=True)
         return country_code
 
+    def set_metadata_from_shipping_address(self, shipping_address):
+        from ..graphql.meta.mutations.utils import save_instance
+        from ..graphql.meta.types import get_valid_metadata_instance
```

Comment:
hm, that way I will need to think of putting this method to a separate file or a helper file - this imports were made due to circular imports - but now I know I should avoid this. Do you have any suggestions where to put this set method? or is there any rule to follow?

Reply:
I see that currently is used only in `CheckoutShippingAddressUpdate` mutation, so it could be the class method or in `saleor/graphql/checkout/mutations/utils.py` file. 

If you move it to `CheckoutShippingAddressUpdate` and make a class method, you will be able to use the `BaseMutation` methods: `validate_and_update_metadata` and `validate_metadata_keys`.